File: April 2022/April_17.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode increasingBST(TreeNode root) {
        inOrder(root);
        return ans;
    }
    
    TreeNode curr = null, ans = null;
    
    private void inOrder(TreeNode root){
        if(root == null)
            return;
        inOrder(root.left);
        root.left = null;
        if(curr != null)
            curr.right = root;
        else
            ans = root;
        curr = root;
        inOrder(root.right);
    }
}

File: April 2022/April_17.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* curr = nullptr;
    TreeNode* ans = nullptr;
    
    void inOrder(TreeNode* root){
        if(root == nullptr)
            return;
        inOrder(root->left);
        root->left = nullptr;
        if(curr != nullptr)
            curr->right = root;
        else
            ans = root;
        curr = root;
        inOrder(root->right);
    }
    
    TreeNode* increasingBST(TreeNode* root) {
        inOrder(root);
        return ans;
    }
};
